p2p/net/conn_test.go

```diff
@@ -25,8 +25,8 @@ func TestSendReceiveMessage(t *testing.T) {
 	rwcam := NewReadWriteCloseAddresserMock()
 	rPub := generatePublicKey()
 	formatter := delimited.NewChan(10)
-	conn := newConnection(rwcam, netw, formatter, rPub, netw.logger)
-	conn.SetSession(&NetworkSessionImpl{})
+	conn := newConnection(rwcam, netw, formatter, rPub, nil, netw.logger)
+	conn.SetSession(&networkSessionImpl{})
```

Comment:
since you added `session` to the construction on a connection, maybe SetSession isn't needed, but why pass null if you could just pass `&networkSessionImpl` instead

Reply:
Missed it, fixed now.